src/economy.cpp

```diff
@@ -914,6 +914,14 @@ void StartupEconomy()
 	_economy.infl_amount_pr = max(0, _settings_game.difficulty.initial_interest - 1);
 	_economy.fluct = GB(Random(), 0, 8) + 168;
 
+	if (_settings_game.economy.inflation) {
+		/* Apply inflation that happened before our game start year. */
+		int months = (_cur_year - ORIGINAL_BASE_YEAR) * 12;
```

Comment:
This may need a min(_cur_year,  ORIGINAL_MAX_YEAR), possibly with some +/- 1.

Reply:
Yep, it does.